File: src/lib/stores/project.ts
import { writable, type Writable } from 'svelte/store';
import type { Project } from '$lib/types/project';

// Project store - holds current project state
export const project: Writable<Project | null> = writable(null);

// Helper functions for project operations
export function createProject(name: string) {
  // TODO: Call Tauri command to create new project
  console.log('Creating project:', name);
}

export function saveProject(path: string) {
  // TODO: Call Tauri command to save project
  console.log('Saving project to:', path);
}

export function loadProject(path: string) {
  // TODO: Call Tauri command to load project
  console.log('Loading project from:', path);
}

export function markProjectModified() {
  project.update((p) => {
    if (p) {
      return {
        ...p,
        modified_at: new Date().toISOString(),
      };
    }
    return p;
  });
}

File: src-tauri/src/commands/project.rs
// Project command stubs - implement these according to tasks.md

#[tauri::command]
pub async fn create_new_project(name: String) -> Result<String, String> {
    // TODO: T104 - Implement project creation
    Err("Not implemented yet".to_string())
}

#[tauri::command]
pub async fn save_project(path: String) -> Result<String, String> {
    // TODO: T102 - Implement project save
    Err("Not implemented yet".to_string())
}

#[tauri::command]
pub async fn load_project(path: String) -> Result<String, String> {
    // TODO: T103 - Implement project load
    Err("Not implemented yet".to_string())
}

